libsolidity/parsing/Parser.cpp

```diff
@@ -322,11 +322,18 @@ StateMutability Parser::parseStateMutability(Token::Value _token)
 	StateMutability stateMutability(StateMutability::NonPayable);
 	if (_token == Token::Payable)
 		stateMutability = StateMutability::Payable;
-	// FIXME: constant should be removed at the next breaking release
-	else if (_token == Token::View || _token == Token::Constant)
+	else if (_token == Token::View)
 		stateMutability = StateMutability::View;
 	else if (_token == Token::Pure)
 		stateMutability = StateMutability::Pure;
+	else if (_token == Token::Constant)
+	{
+		stateMutability = StateMutability::View;
```

Comment:
Why to set this if the throw an exception?

Reply:
Well on another note, why isn't the below a fatal error?

Reply:
Because it does not have to.